### ISSUE TITLE
Simple API to run load and evaluate  datasets based on tasks and input data

### ISSUE THREAD

**Issue by yoavkatz** [IBM/unitxt#1327]
We need a simple way for a user to load a Unitxt dataset based on existing files or data structures.
This is required for ilab (@Roni-Friedman )  and also for other use cases.

I thought that maybe a dataframe is a common interface that everyone knows. 

load_dataset(task="tasks.classification.multi_class",  test_set = pd.from_csv("test.csv"))
load_dataset(task="tasks.classification.multi_class", test_set = pd.from_csv("test.csv"), train_set=pd.from_csv("train.csv" )

And

evaluate(predications, task="tasks.classification.multi_class", test_set = pd.from_csv("test.csv"))

The assumption is that the fields of input dataframe  are the same as the task, and a clear error message will be presented if not.

**Comment by elronbandel** [comment]
Is `unitxt.api.produce` is not enough? you could use it with `produce(pd.from_csv("test.csv").to_list(), task="your_task")`
https://github.com/IBM/unitxt/blob/582d96f87cbfcec4367a4d8ed37d05669ebba265/src/unitxt/api.py#L132-L139

**Comment by yoavkatz** [comment]
I  did not even think about produce.  I'll check.  It has to be more prominently documented (Currently it's under "Production and Inference").

**Comment by yoavkatz** [comment]
I tried:

# Set up question answer pairs in a dictionary
data = [
        {"question": "What is the capital of Texas?", "answers": ["Austin"]},
        {"question": "What is the color of the sky?", "answers": ["Blue"]},
]

test_dataset = produce(data,task="tasks.qa.open", template="templates.qa.open.title")

Put the output dataset does not contain references field, so it can be evaluated.

**Comment by elronbandel** [comment]
You are right. I forgot it was meant for inference. We should document it properly.
